lib/components/viewers/stop-viewer.js

```diff
@@ -285,20 +285,26 @@ class PatternRow extends Component {
       timeFormat
     } = this.props
     // sort stop times by next departure
-    let sortedStopTimes = null
-    if (stopTimes) {
+    let sortedStopTimes = []
+    const hasStopTimes = stopTimes && stopTimes.length > 0
+    if (hasStopTimes) {
       sortedStopTimes = stopTimes.sort((a, b) => {
```

Comment:
The [Array.sort](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort) method sorts an array in-place, so this reassignment is not needed.

Reply:
Well, I guess this reassignement sort of is needed in order to not reassign a const. Still seems kind of like a bad practice to be manipulating a variable that is a property incoming from a parent component.

Reply:
What if we just concat the array before sorting?